chore(main): add documentation for Windows (WSL2) installation (#1045)
instill-ai/instill-core: README.md

## Prerequisites

- **macOS or Linux** - **🔮 Instill Core** works on macOS or Linux

- **Windows** - **🔮 Instill Core** works on Windows through Windows Linux Subsystem (WSL2)
  

>   *WSL specific settings*:
>   
>   (a) The package `yq` is not installed on Ubuntu WSL2 by default, install the lastest version of `yq` from the [official Github Repository](https://github.com/mikefarah/yq).
>   
>   (b) Install the latest version of Docker Desktop on Windows and enable the WSL2 integration following [this tutorial from Microsoft](https://docs.docker.com/desktop/wsl/)
>   
>   (c) Install `cuda-toolkit` on WSL2 following [this tutorial from Nvidia](https://docs.nvidia.com/cuda/wsl-user-guide/index.html#getting-started-with-cuda-on-wsl-2) 

Reviewer: ```suggestion
  - Install the lastest version of `yq` from the GitHub [Repository](https://github.com/mikefarah/yq), as the package `yq` is not installed on Ubuntu WSL2 by default
  
  - Install the latest version of Docker Desktop on Windows and enable the WSL2 integration following the [tutorial](https://docs.docker.com/desktop/wsl) by Docker
  
  - (optional) Install `cuda-toolkit` on WSL2 following the [tutorial](https://docs.nvidia.com/cuda/wsl-user-guide/index.html#getting-started-with-cuda-on-wsl-2) by NVIDIA
```

Author: Changed as requested.
